### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by YogyaGamage**
### Purpose
Update mockito 1 to mockito 2 to support JDK 11 and JDK 17 compilation.

### When should this PR be merged
After this PR is merged to the main branch, we can't build the branch on JDK 8

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

File: components/org.wso2.carbon.identity.query.saml/pom.xml [hunk 1]
```diff
@@ -229,6 +229,11 @@
 			<artifactId>org.apache.felix.scr.ds-annotations</artifactId>
 			<scope>provided</scope>
 		</dependency>
+		<dependency>
+            <groupId>com.sun.xml.parsers</groupId>
+            <artifactId>jaxp-ri</artifactId>
+			<scope>test</scope>
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.query.saml/pom.xml:235]: indentation issue

File: components/org.wso2.carbon.identity.query.saml/pom.xml [hunk 2]
```diff
@@ -265,6 +270,28 @@
 					<systemPropertyVariables>
 						<jacoco-agent.destfile>target/jacoco.exec</jacoco-agent.destfile>
 					</systemPropertyVariables>
+					<argLine>
+                        --add-opens=java.base/java.lang=ALL-UNNAMED
+						--add-opens=java.base/java.lang.invoke=ALL-UNNAMED
+                        --add-opens=java.base/java.util=ALL-UNNAMED
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.query.saml/pom.xml:276]: indentation issue

File: components/org.wso2.carbon.identity.sso.saml/pom.xml [hunk 3]
```diff
@@ -420,6 +434,30 @@
...
                     </suiteXmlFiles>
+                    <argLine>
+                        --add-opens=java.base/java.lang=ALL-UNNAMED
+                        --add-opens=java.base/java.lang.invoke=ALL-UNNAMED
+                        --add-opens=java.base/java.util=ALL-UNNAMED
+                        --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
+						--add-opens=java.base/java.util.zip=ALL-UNNAMED
+                        --add-opens=java.base/java.util.random=ALL-UNNAMED
+                        --add-opens=java.base/java.security=ALL-UNNAMED
+						--add-opens=java.base/java.io=ALL-UNNAMED
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.sso.saml/pom.xml:445]: fix indentation issue in all places

File: components/org.wso2.carbon.identity.sso.saml/pom.xml [hunk 4]
```diff
@@ -301,6 +298,23 @@
             <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
             <scope>provided</scope>
         </dependency>
+
+        <!-- for Java 17 Compatibility -->
+        <dependency>
+            <groupId>com.sun.xml.ws</groupId>
+            <artifactId>jaxws-ri</artifactId>
+            <type>pom</type>
```
[the remark this answers is not in the document]
Comment (YogyaGamage) [reply components/org.wso2.carbon.identity.sso.saml/pom.xml:306]: > Why the type is POM here?

The type "pom" is required for the dependency. According to [Licence Concerns when updating the products to Java 11] we can use the dependency in WSO2 products.

**Review by piraveena — APPROVED** [review]

**Pull request merged**